sympy/core/tests/test_function.py

```diff
@@ -859,6 +859,13 @@ def test_issue_12005():
     assert e5.diff(x) == Derivative(f(x), x, x)
     assert f(g(x)).diff(g(x), g(x)) == Subs(Derivative(f(y), y, y), (y,), (g(x),))
 
+def test_issue_13843():
+    x = symbols('x')
+    f = Function('f')
+    m, n = symbols('m n', integer=True)
```

Comment:
What if you remove the `integer=True` condition?

Reply:
```
>>> m, n = symbols('m n')
>>> Derivative(Derivative(f(x), (x, m+5)), (x, n+3))
Derivative(Derivative(f(x), (x, m + 5)), (x, n + 3))
```